src/standalone/Vegetation/soil_moisture_stress.jl

```diff
@@ -0,0 +1,390 @@
+        )
+    end
+    if β0 <= 0
+        throw(
+            ArgumentError(
+                "Intrinsic moisture stress factor `β0` must be greater than zero",
+            ),
+        )
+    end
+
+    c = FT(c)
+    β0 = FT(β0)
+    ψ_fc = FT(ψ_fc)
+    ψ_wp = FT(ψ_wp)
+
+    # Calculate effective saturation at field capacity and wilting point
+    if hydrology_cm isa ClimaCore.Fields.Field
+        S_fc = @. ClimaLand.Soil.inverse_matric_potential(hydrology_cm, ψ_fc)
+        S_wp = @. ClimaLand.Soil.inverse_matric_potential(hydrology_cm, ψ_wp)
+        θ_high = @. S_fc * (ν - θ_r) + θ_r
+        θ_low = @. S_wp * (ν - θ_r) + θ_r
+    else
+        S_fc = ClimaLand.Soil.inverse_matric_potential(hydrology_cm, ψ_fc)
+        S_wp = ClimaLand.Soil.inverse_matric_potential(hydrology_cm, ψ_wp)
+        θ_high = S_fc * (ν - θ_r) + θ_r
+        θ_low = S_wp * (ν - θ_r) + θ_r
+    end
+
+    return PiecewiseMoistureStressParameters(
+        θ_high = θ_high,
+        θ_low = θ_low,
+        c = c,
+        β0 = β0,
+        soil_zmin = soil_zmin,
+    )
+end
+
+"""
+    compute_piecewise_moisture_stress(
+        θ_high::FT,
+        θ_low::FT,
+        c::FT,
+        β0::FT,
+        θ::FT,
+    ) where {FT}
+
+This function computes at a point the soil moisture stress factor using the volumetric water content θ (m^3/m^3)
+and four parameters: `θ_high` (field capacity, m^3/m^3), `θ_low` (wilting point, m^3/m^3), `c` (curvature parameter,
+unitless), and `β0` (intrinsic moisture stress factor, unitless). This is a modification to
+the piecewise formulation defined in Egea et al. (2011) where the entire functional form is multiplied by
+`β0` to allow for some "intrinsic moisture stress" even in well-watered conditions.
+
+Citation: https://doi.org/10.1016/j.agrformet.2011.05.019
+"""
+function compute_piecewise_moisture_stress(
+    θ_high::FT,
+    θ_low::FT,
+    c::FT,
+    β0::FT,
+    θ::FT,
+) where {FT}
+    # avoid taking e.g. sqrt of negative numbers for rational c
+    arg = max((θ - θ_low) / (θ_high - θ_low), FT(0))
+
+    return β0 * max(FT(0), min(FT(1), arg^c))
+end
+
+"""
+    update_soil_moisture_stress!(
+        p,
+        Y,
+        model::PiecewiseMoistureStressModel,
+        canopy,
+    )
+
+This updates the soil moisture stress factor according to the piecewise soil moisture stress model.
+"""
+function update_soil_moisture_stress!(
+    p,
+    Y,
+    model::PiecewiseMoistureStressModel,
+    canopy,
+)
+    # compute root-integrated soil water content
+    if :soil in canopy.boundary_conditions.prognostic_land_components
+        # unpack soil moisture field, parameters
+        ϑ_l = Y.soil.ϑ_l
+        (; θ_high, θ_low, c, β0, soil_zmin) = model.parameters
+        z = ClimaCore.Fields.coordinate_field(axes(ϑ_l)).z
+
+        # normalized distribution for root density
+        norm = @. lazy(
+            Canopy.PlantHydraulics.root_distribution_CDF(
+                soil_zmin,
```

Comment:
I think here it makes sense to use the minimum of the soil domain

Reply:
because the zmin we use has to be the same as in the integrals below for the compute norm to be the actual norm

Reply:
ah, this is hard to obtain from "canopy" alone quantities. I changed this to compute the norm by doing the integral, even though it is analytically available, and constant in time.